contracts/utils/SignatureValidator.sol

```diff
@@ -1,195 +1,91 @@
-        bytes32 hash,
-        address walletAddress,
-        bytes memory signature
-    ) internal view returns (bool isValid) {
-        bytes memory _calldata = abi.encodeWithSelector(IWallet(walletAddress).isValidSignature.selector, hash, signature);
-        bytes32 magic_salt = bytes32(bytes4(keccak256("isValidWalletSignature(bytes32,address,bytes)")));
-        assembly {
-            if iszero(extcodesize(walletAddress)) {
-                // Revert with `Error("WALLET_ERROR")`
-                mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
-                mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
-                mstore(64, 0x0000000c57414c4c45545f4552524f5200000000000000000000000000000000)
-                mstore(96, 0)
-                revert(0, 100)
-            }
-
-            let cdStart := add(_calldata, 32)
-            let success := staticcall(
-                gas(), // forward all gas
-                walletAddress, // address of Wallet contract
-                cdStart, // pointer to start of input
-                mload(_calldata), // length of input
-                cdStart, // write output over input
-                32 // output size is 32 bytes
-            )
-
-            if iszero(eq(returndatasize(), 32)) {
-                // Revert with `Error("WALLET_ERROR")`
-                mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
-                mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
-                mstore(64, 0x0000000c57414c4c45545f4552524f5200000000000000000000000000000000)
-                mstore(96, 0)
-                revert(0, 100)
-            }
-
-            switch success
-            case 0 {
-                // Revert with `Error("WALLET_ERROR")`
-                mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
-                mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
-                mstore(64, 0x0000000c57414c4c45545f4552524f5200000000000000000000000000000000)
-                mstore(96, 0)
-                revert(0, 100)
-            }
-            case 1 {
-                // Signature is valid if call did not revert and returned true
-                isValid := eq(
-                    and(mload(cdStart), 0xffffffff00000000000000000000000000000000000000000000000000000000),
-                    and(magic_salt, 0xffffffff00000000000000000000000000000000000000000000000000000000)
-                )
-            }
-        }
-        return isValid;
-    }
+    // Anything else is illegal (We do not return false because
+    // the signature may actually be valid, just not in a format
+    // that we currently support. In this case returning false
+    // may lead the caller to incorrectly believe that the
+    // signature was invalid.)
+    revert("SignatureValidator: unsupported signature");
```

Comment:
This will never be reached because we check `signatureTypeRaw <= uint8(SignatureType.ZX1271)`?

Reply:
The `SignatureType .Invalid` will reach here.

Reply:
Maybe can remove `SignatureType.Illegal` case?

Reply:
Since `Illegal` also serves the purpose of representing the default value of enum type, maybe we can remove `Invalid` instead? (This final revert is implicitly the case for `Invalid` anyway.)

Reply:
The goal is not changing the enum order. But if you're talking about `if (signatureType == SignatureType.Illegal)` case, yeah I was thinking about removing this.

Reply:
> yeah I was thinking about removing this.

Yeah that would make sense. Though I'm still not fully understand the difference between `Illegal` and `Invalid`.🧐

Reply:
After discussion, fixed in 472d85b0f54876e716461f8633e9cbcc3da1e16e